src/simple_regular_solve.jl

```diff
@@ -50,6 +50,136 @@ function DiffEqBase.solve(jump_prob::JumpProblem, alg::SimpleTauLeaping;
         interp = DiffEqBase.ConstantInterpolation(t, u))
 end
 
+struct SimpleAdaptiveTauLeaping <: DiffEqBase.DEAlgorithm
+    epsilon::Float64  # Error control parameter
+end
+
+SimpleAdaptiveTauLeaping(; epsilon=0.05) = SimpleAdaptiveTauLeaping(epsilon)
+
+function compute_gi(u, nu, hor, i)
+    max_order = 1.0
+    for j in 1:size(nu, 2)
+        if abs(nu[i, j]) > 0
+            max_order = max(max_order, Float64(hor[j]))
+        end
+    end
+    return max_order
```

Comment:
This isn't necessarily the correct value for $g_i$ though when `hor[i] > 1`. If `hor[i] == 2` you need to precalculate if any reaction for which species `i` is a substrate has the substrate stoichoimetry of 2 or more (using `reactant_stoch` for example). Then you need to use the formulate for the updated `g_i` in case (II). Similar for if `hor[i] ==3`. 

We don't need to support if `hor[i] > 3`, but the solver should probably error if it detects this is the case.